packages/perseus/src/validation.types.ts

```diff
@@ -236,58 +238,105 @@ export type PerseusTableRubric = {
-    | PerseusMatcherRubric
-    | PerseusMatrixRubric
-    | PerseusNumberLineScoringData
-    | PerseusNumericInputRubric
-    | PerseusOrdererRubric
-    | PerseusPlotterScoringData
-    | PerseusRadioRubric
-    | PerseusSorterRubric
-    | PerseusTableRubric;
-
-export type UserInput =
-    | PerseusCategorizerUserInput
-    | PerseusCSProgramUserInput
-    | PerseusDropdownUserInput
-    | PerseusExpressionUserInput
-    | PerseusGrapherUserInput
-    | PerseusGroupUserInput
-    | PerseusIFrameUserInput
-    | PerseusInputNumberUserInput
-    | PerseusInteractiveGraphUserInput
-    | PerseusLabelImageUserInput
-    | PerseusMatcherUserInput
-    | PerseusMatrixUserInput
-    | PerseusNumberLineUserInput
-    | PerseusNumericInputUserInput
-    | PerseusOrdererUserInput
-    | PerseusPlotterUserInput
-    | PerseusRadioUserInput
-    | PerseusSorterUserInput
-    | PerseusTableUserInput;
-
-export type UserInputMap = {[widgetId: string]: UserInput};
+export interface RubricRegistry {
+    categorizer: PerseusCategorizerScoringData;
+    "cs-program": PerseusCSProgramRubric;
+    // definition: PerseusDefinitionScoringData;
+    dropdown: PerseusDropdownRubric;
+    // explanation: PerseusExplanationScoringData;
+    expression: PerseusExpressionRubric;
+    grapher: PerseusGrapherRubric;
+    "graded-group-set": PerseusGradedGroupSetRubric;
+    "graded-group": PerseusGradedGroupRubric;
+    group: PerseusGroupRubric;
+    iframe: PerseusIFrameRubric;
+    image: PerseusLabelImageRubric;
+    "input-number": PerseusInputNumberRubric;
+    // interaction: PerseusInteractionScoringData;
+    "interactive-graph": PerseusInteractiveGraphRubric;
+    "label-image": PerseusLabelImageRubric;
+    matcher: PerseusMatcherRubric;
+    matrix: PerseusMatrixRubric;
+    // measurer: PerseusMeasurerRubric;
+    // "molecule-renderer": PerseusMoleculeRendererRubric;
+    "number-line": PerseusNumberLineScoringData;
+    "numeric-input": PerseusNumericInputRubric;
+    orderer: PerseusOrdererRubric;
+    // "passage-ref-target": PerseusRefTargetRubric;
+    // "passage-ref": PerseusPassageRefRubric;
+    // passage: PerseusPassageRubric;
+    // "phet-simulation": PerseusPhetSimulationRubric;
+    // "python-program": PerseusPythonProgramRubric;
+    plotter: PerseusPlotterScoringData;
+    radio: PerseusRadioRubric;
+    sorter: PerseusSorterRubric;
+    table: PerseusTableRubric;
+    // video: PerseusVideoRubric;
+
+    // Deprecated widgets
+    // sequence: PerseusAutoCorrectRubric;
+}
+
+/**
+ * A map of scoring data (previously referred to as "rubric"), keyed by
+ * `widgetId`. This data is used to score a learner's guess for a PerseusItem.
+ *
+ * NOTE:  The value in this map is intentionally a subset of WidgetOptions<T>.
+ * By using the same shape (minus any unneeded render data), we are able to
+ * share functionality that understands how to traverse maps of `widget id` to
+ * `options`.
+ */
+export type RubricMap = {
```

Comment:
So overall the point of this PR seems to be aimed at more closely matching widget-specific data to the widget ID?

Doesn't this reinforce the thing you and Nicole were against - tightly binding the widget `type` and the widget `id`? Like if we were to decide tomorrow that we were going to give all widgets UUIDs (which we should as we've seen in the GradedGroup issue), wouldn't that break this code?

I'm having trouble aligning the description of the PR and the ticket with the actual changes. This is like pre/post work for LEMS-2561?

Reply:
Sorry, this PR drifted from what I initially typed out as the PR description. I'll update the description. It's basically type work to make the types we already use (full widget options, Scoring Data, Validation Data) a bit more clear and well-typed. 

I agree that I don't like us buying further into the mapping of widget ID to data in our various maps. Maybe we can chat about that in a future PR? I'm also happy to have pushback and do more work in this PR to isolate the Renderer and outer code knowing so much about widget option details.